packages/browser-repl/src/components/shell.tsx

```diff
@@ -127,13 +127,17 @@ interface ShellProps {
    */
   initialHistory: readonly string[];
 
+  /**
+   * A boolean so we can keep the isOperationInProgress state between sessions.
```

Comment:
If the expectation is that `onOperationStarted` and `onOperationEnd` are used to keep track of this state outside of the shell component itself, then I'd mention that in the doc comment here.

I'm wondering, though, if it doesn't make sense to just pass this prop as the initial value of `operationInProgress` and keep tracking this in the state of the shell itself, semantically that definitely seems like the right thing to do.

If you decided against that, you'll also want to update `src/sandbox.tsx` to adopt the new way of keeping track of this state.

Reply:
> and keep tracking this in the state of the shell itself

I'm not following? Where do you mean in the shell? The bug here is that we lose that variable when the shell re-renders, so I'm making it a controlled prop.

Reply:
(I should definitely add some comments to document this)

Reply:
Right, I'm saying, if we expect the parent of this component (= `Shell`) to keep track of the state, then that should be documented here and we should adjust the sandbox example as well

But the real underlying issue seems to be that all shell state gets lost in this case, that's why we have the `initial...` properties on the component, and I'm wondering why we treat this particular property differently (i.e. don't also provide an `initialOperationInProgress` and use that to set the default value of the flag in the Shell component state)

Reply:
My thinking is that if we keep track of initialOperationInProgress to exactly match the internal operationInProgress state at all times because at any point the component could be replaced, then why have both? Same actually for the other props and state this component has that acts like that. Tempting to just change them all and turn this into a function component while at it.

Reply:
I changed inputText, output, history and isOperationInProgress into controlled props and updated both the README and sandbox accordingly. While at it I just changed the whole thing into a function component.

I have to basically rewrite the tests now because they were still using enzyme so I just removed them for the moment. Will replace in a separate commit as I think it is easier that way around at this stage.